cpp/arcticdb/processing/clause.cpp

```diff
@@ -684,6 +686,26 @@ void RowRangeClause::set_processing_config(const ProcessingConfig& processing_co
                 end_ = total_rows;
             }
             break;
+        case RowRangeType::RANGE:
+            // Wrap around negative indices.
+            start_ = (
+                user_provided_start_ >= 0 ?
+                std::min(user_provided_start_, total_rows) :
+                std::max(total_rows + user_provided_start_, static_cast<int64_t>(0))
+            );
+            end_ = (
+                user_provided_end_ >= 0 ?
+                std::min(user_provided_end_, total_rows) :
+                std::max(total_rows + user_provided_end_ + 1, static_cast<int64_t>(0))
+            );
+            if (start_ > end_) {
+                internal::raise<ErrorCode::E_ASSERTION_FAILURE>(
+                        "RowRangeClause start index {} is greater than end index {}; originally (start, end)=({}, {}) ",
+                        start_, end_, user_provided_start_, user_provided_end_);
+            }
+            n_ = end_ - start_;
```

Comment:
Note to reviewers: Should we also port this logic on the Python side [here](https://github.com/man-group/ArcticDB/pull/864/files#diff-fd55a7274e8a49f65e2f62074974d20809bde05f639f4d55fbe7f8430f772358R135-R142)?

Reply:
Is this comment out of date? The link doesn't reference any Python code for me

Reply:
Yes, it is out of date.